src/chainer.test.js

```diff
@@ -1 +1,60 @@
+'use strict';
+
 const { chainer } = require('./chainer');
+
+describe(`'Chainer' function`, () => {
+  let f1;
+  let f2;
+  let f3;
+
+  beforeEach(() => {
+    f1 = jest.fn(x => x - 5);
+    f2 = jest.fn(x => x + 2);
+    f3 = jest.fn(x => x * 3);
+  });
+
+  it(`should be declared`, () => {
+    expect(chainer).toBeInstanceOf(Function);
+  });
+
+  it(`should return a function`, () => {
+    const args = [() => {}, () => {}, () => {}];
+
+    expect(chainer(args)).toBeInstanceOf(Function);
+  });
+
+  it(`should return 'NaN' if one of the 'chainer'`
+  + ` parameters is 'undefined'`, () => {
+    f1 = () => undefined;
+
+    expect(chainer([f1, f2, f3])(4)).toBeNaN();
+  });
+
+  it(`should return callback that return value`, () => {
+    const returnedCb = jest.fn().mockReturnValue(x => x);
+    const result = chainer([returnedCb()])(3);
+
+    expect(result).not.toBeUndefined();
+  });
+
+  it(`should invoke each function from 'functions'`, () => {
+    chainer([f1, f2, f3])();
+
+    expect(f1).toHaveBeenCalledTimes(1);
+    expect(f2).toHaveBeenCalledTimes(1);
+    expect(f3).toHaveBeenCalledTimes(1);
+  });
+
+  it(`result of callback should consist of the current call`
+  + ` that invokes the previous one as a parameter`, () => {
+    chainer([f1, f2, f3])(5);
+
+    expect(f2).toHaveBeenCalledWith(0);
+  });
```

Comment:
The test case expects `f2` to be called with `0`, but this assumes that `f1` applied to `5` results in `0`. Given `f1 = jest.fn(x => x - 5)`, this is correct. Ensure that this logic aligns with the intended functionality of the `chainer` function.